cpp/src/community/k_truss_impl.cuh

```diff
@@ -0,0 +1,916 @@
+        thrust::make_counting_iterator<edge_t>(num_invalid_edges),
+        [num_triangles = raft::device_span<edge_t>(num_triangles.data() + num_valid_edges, num_invalid_edges), intersection_offsets = raft::device_span<size_t const>(intersection_offsets.data(), intersection_offsets.size())]__device__(auto i) {
+    
+          num_triangles[i] -= intersection_offsets[i + 1] - intersection_offsets[i];
+        });
+
+      auto pair_ = thrust::make_tuple(6, 1);
+      auto itr_ = thrust::find(edge_first + num_valid_edges, edge_first + edgelist_srcs.size(), pair_);
+      
+      auto dist_ = thrust::distance(edge_first + num_valid_edges, itr_);
+
+      // FIXME: Find a way to not have to maintain a dataframe_buffer
+      auto vertex_pair_buffer_p_r_edge_p_q =
+        allocate_dataframe_buffer<thrust::tuple<vertex_t, vertex_t>>(intersection_indices.size(),
+                                                                     handle.get_stream());
+
+      thrust::tabulate(
+        handle.get_thrust_policy(),
+        get_dataframe_buffer_begin(vertex_pair_buffer_p_r_edge_p_q),
+        get_dataframe_buffer_end(vertex_pair_buffer_p_r_edge_p_q),
+        generate_p_r_q_r<vertex_t, edge_t, true>{
+          raft::device_span<size_t const>(intersection_offsets.data(), intersection_offsets.size()),
+          raft::device_span<vertex_t const>(intersection_indices.data(),
+                                            intersection_indices.size()),
+          raft::device_span<vertex_t>(edgelist_srcs.data() + num_valid_edges, num_invalid_edges),
+          raft::device_span<vertex_t>(edgelist_dsts.data() + num_valid_edges, num_invalid_edges)});
+
+      auto vertex_pair_buffer_q_r_edge_p_q =
+        allocate_dataframe_buffer<thrust::tuple<vertex_t, vertex_t>>(intersection_indices.size(),
+                                                                     handle.get_stream());
+
+      thrust::tabulate(
+        handle.get_thrust_policy(),
+        get_dataframe_buffer_begin(vertex_pair_buffer_q_r_edge_p_q),
+        get_dataframe_buffer_end(vertex_pair_buffer_q_r_edge_p_q),
+        generate_p_r_q_r<vertex_t, edge_t, false>{
+          raft::device_span<size_t const>(intersection_offsets.data(), intersection_offsets.size()),
+          raft::device_span<vertex_t const>(intersection_indices.data(),
+                                            intersection_indices.size()),
+          raft::device_span<vertex_t>(edgelist_srcs.data() + num_valid_edges, num_invalid_edges),
+          raft::device_span<vertex_t>(edgelist_dsts.data() + num_valid_edges, num_invalid_edges)});
+
+      // Unrolling the edges require the edges to be sorted by destination
+      // re-sort the invalid edges by 'dst'
+      thrust::sort_by_key(handle.get_thrust_policy(),
+                          transposed_edge_first + num_valid_edges,
+                          transposed_edge_first + edgelist_srcs.size(),
+                          num_triangles.begin() + num_valid_edges);
```

Comment:
I don't think I can because my `intersection offsets` and `indices` are derived from an edge list sorted by `src`. Unless I am done processing the `intersection offsets` and indices, I can't re-sort the edge list by `dst`

Reply:
Oh, I see, yes, you're right